packages/node-provider/src/node-provider.ts

```diff
@@ -1,7 +1,35 @@
-import { INodeProvider, NodeMessage } from "./types";
+import {
+  INodeProvider,
+  NodeMessage,
+  NodeMessageReceivedCallback,
+  NodeMessageType,
+  NodeProviderStatus
+} from "./types";
 
 export default class NodeProvider implements INodeProvider {
-  public onMessage(callback: (message: NodeMessage) => void) {}
+  private internalStatus: NodeProviderStatus = NodeProviderStatus.OFFLINE;
+
+  constructor() {}
+
+  public onMessage(callback: NodeMessageReceivedCallback) {}
```

Comment:
Should we support subscribing to any incoming message with `onMessage` and to specific message types with `on/once`?

Reply:
I think we can just go with `onMessage` for now. Reviewing our usage of the NodeProvider in CF.js makes me think it's unnecessary at this point for us to have to subscribe callbacks to specific message types rather than just have one we can dispatch from based on type.

Reply:
`on/once` also requires extra maintenance to ensure CFjs subscribes to all the right message types. 

Instead, CFjs can (and will) expose `on/once` convenience methods.